projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.spec.ts

```diff
@@ -43,4 +51,184 @@ describe('SelectedCaseComponent', () => {
+        }]
+    }];
+    component.shareCases$ = of(shareCases);
+    fixture.detectChanges();
+
+    user = {
+      idamId: 'U111111',
+      firstName: 'James',
+      lastName: 'Priest',
+      email: 'james.priest@test.com'
+    };
+   expect(component.canRemove('C111111', user)).toEqual(true);
+  });
+
+  it('case cannot be cancelled', () => {
+    shareCases = [{
+      caseId: 'C111111',
+      caseTitle: 'Sarah vs Pete',
+      sharedWith: [{
+        idamId: 'U111111',
+        firstName: 'James',
+        lastName: 'Priest',
+        email: 'james.priest@test.com'
+      }],
+      pendingShares: [{
+        idamId: 'ps111111',
+        firstName: 'JamesPS',
+        lastName: 'PriestPS',
+        email: 'jamesps.priestps@test.com'
+      }],
+      pendingUnshares: [{
+        idamId: 'pus111111',
+        firstName: 'JamesPUS',
+        lastName: 'PriestPUS',
+        email: 'jamespus.priestpus@test.com'
+      }],
+    }];
+    component.shareCases$ = of(shareCases);
+    fixture.detectChanges();
+    user = {
+      idamId: 'pus111111',
+      firstName: 'JamesPUS',
+      lastName: 'PriestPUS',
+      email: 'jamespus.priestpus@test.com'
+    };
+
+    expect(component.canCancel('C111111', user)).toEqual(true);
+  });
+
+  it('case can be cancelled', () => {
+    shareCases = [{
+      caseId: 'C111111',
+      caseTitle: 'Sarah vs Pete',
+      sharedWith: [{
+        idamId: 'U111111',
+        firstName: 'James',
+        lastName: 'Priest',
+        email: 'james.priest@test.com'
+      }],
+      pendingShares: [{
+        idamId: 'ps111111',
+        firstName: 'JamesPS',
+        lastName: 'PriestPS',
+        email: 'jamesps.priestps@test.com'
+      }],
+      pendingUnshares: [{
+        idamId: 'pus111111',
+        firstName: 'JamesPUS',
+        lastName: 'PriestPUS',
+        email: 'jamespus.priestpus@test.com'
+      }],
+    }];
+    component.shareCases$ = of(shareCases);
+    fixture.detectChanges();
+    user = {
+      idamId: 'ps111111',
+      firstName: 'JamesPS',
+      lastName: 'PriestPS',
+      email: 'jamesps.priestps@test.com'
+    };
+
+    expect(component.canCancel('C111111', user)).toEqual(true);
+  });
+
+  it('checking for isToBeRemoved', () => {
+    shareCases = [{
+      caseId: 'C111111',
+      caseTitle: 'Sarah vs Pete',
+      sharedWith: [{
+        idamId: 'U111111',
+        firstName: 'James',
+        lastName: 'Priest',
+        email: 'james.priest@test.com'
+      }],
+
+      pendingUnshares: [{
+        idamId: 'pus111111',
+        firstName: 'JamesPUS',
+        lastName: 'PriestPUS',
+        email: 'jamespus.priestpus@test.com'
+      }],
+    }];
+    component.shareCases$ = of(shareCases);
+    fixture.detectChanges();
+    user = {
+      idamId: 'pus111111',
+      firstName: 'JamesPUS',
+      lastName: 'PriestPUS',
+      email: 'jamespus.priestpus@test.com'
+    };
+
+    expect(component.isToBeRemoved('C111111', user)).toEqual(true);
+  });
+
+
```

Comment:
Please add tests to verify html template and events from template,

1. if users listed and correct number of users displayed
2. If Remove label displayed
3. If Cancel label displayed
4. If to be removed label displayed
5. If to be added label is displayed
6. If onRemove triggered on clicking Remove link
7. If onCancel triggered on clicking cancel link